packages/devtools_server/lib/src/server.dart

```diff
@@ -476,6 +456,92 @@ Future<void> registerLaunchDevToolsService(
   }
 }
 
+Future<Map<String, dynamic>> launchDevTools(
+    Map<String, dynamic> params,
+    Uri vmServiceUri,
+    String devToolsUrl,
+    bool headlessMode,
+    bool machineMode) async {
+  // Prints a launch event to stdout so consumers of the DevTools server
+  // can see when clients are being launched/reused.
+  void emitLaunchEvent({@required bool reused, @required bool notified}) {
```

Comment:
consider moving this out into a private library method in order to shorten the impl of the `launchDevTools` method.

Reply:
Done! (d05a27b7)